### PR TITLE
[W4.5c][W15-A3] Yash Chowdhary

### PR DESCRIPTION
**Pull request by yash-chowdhary**
Application displays appropriate message to user and starts over.

### CODE REVIEW

**Comment by yash-chowdhary** [comment]
Some code adapted from #44 (restarting the application).

**Review by eldriclim — COMMENTED** [review]
Nice, keep up the good work.

**Review by eldriclim — COMMENTED** [review]
Nice, keep up the good work.

**Comment by eldriclim** [comment]
@yash-chowdhary Please close PR after reading comments.

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -105,13 +110,17 @@ private void recordResult(CommandResult result) {
...
+    private CommandResult executeCommand(Command command) throws StorageOperationException {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        } catch (StorageOperationException soe){
+            ui.showToUser(Messages.MESSAGE_READ_ONLY);
+            throw new StorageOperationException(Messages.MESSAGE_READ_ONLY);
```
Comment (eldriclim) [inline src/seedu/addressbook/Main.java:121]: Should the same exception be caught and thrown within the same block?

**Pull request closed without merging**